### DISCUSSION TITLE
开发者例会议题收集（20220628）

### DISCUSSION THREAD

**Discussion by andrewshan** [polarismesh/polaris#461] [Meeting]
例会时间：6月28-29日晚上7点半

议题收集：

1. 北极星性能优化方案讨论 @chuntaojun @shichaoyuan 
- 最终一致性及补偿机制：https://github.com/polarismesh/polaris/issues/429
- 如何向前兼容：https://github.com/polarismesh/polaris/issues/457

2.  TLS相关需求和方案讨论： （下次例会）
- 北极星支持mTLS方案讨论：https://github.com/polarismesh/polaris/issues/422
- discover接口支持TLS：https://github.com/polarismesh/polaris/pull/456
- redis支持TLS：https://github.com/polarismesh/polaris/issues/398

3.  polaris-python设计评审：https://github.com/polarismesh/polaris-python/discussions/3

**Comment by chuntaojun** [comment]
4. 北极星官网结构的调整：https://github.com/polarismesh/polaris/discussions/463

**Comment by saigu** [comment]
5.  sdk中「推空保护」的问题 https://github.com/polarismesh/polaris-java/issues/154

**Comment by andrewshan** [comment]
会议主题：Polaris社区开发者例会 
会议时间：2022/06/28 19:30-20:00 (GMT+08:00) 中国标准时间 - 北京 
点击链接入会，或添加至会议列表： 
https://meeting.tencent.com/dm/UXfjm50TQiTG 
#腾讯会议：719-819-974

**Comment by shichaoyuan** [comment]
6. 单元测试相关的问题
- 单测稳定性，运行时间。 例如 polaris-go 跑一次要三十多分钟，结果不稳定
- 感觉现在普遍是 functional test，而不是 unit test
- 目录结构问题，单独的 test 目录，还是其它？

**Reply by chuntaojun** [reply]
//TODO 测试规范

1、粗糙的测试先 skip
2、优先实现单元测试
3、集成测试后面再统一处理
4、PR要有一定的单元测试覆盖

**Comment by Guo-Zhang** [comment]
# 议题3 - Python SDK提案

PR的message已经详细些了已提交代码的说明。

主要参考我们自己做的Coding SDK：https://github.com/quanttide/coding-devops-sdk-py。

## 架构

- api：使用类似于Coding SDK的模块划分，方便维护各个API。
- config：使用dynaconf。
其他模块根据实际情况决定，和北极星社区讨论。

## DevOps

打包：
- 库：暂时使用setuptools，未来会升级。暂时还没搞明白另外两个库到底有什么优势。
- 配置：pyproject.toml，使用新标准。

测试：
- unittest和pytest，交给社区选择。我们倾向于暂时unittest，和我们这边一起升级到pytest。pytest总体更强大。

CI：
- release触发打包。
- 计划增加master和PR测试。

**Comment by chuntaojun** [comment]
## 本期会议摘要

### 北极星性能优化方案

- 服务端调整实例异步注册的判断开关，通过请求的 Header 中是否携带 async-regis 来判定
- SDK新增RegisterInstance方法，确保能够触发服务端的异步注册逻辑

### 北极星官网结构的调整

- 顶部TAB
  - 首页
  - 项目
  - 文档
  - 博客
  - Github
  - 体验版

- 文档目录
  - 北极星概念
  - 快速入门（快速上手）
  - 使用指南
    - 每个功能点、参数、配置，解释、可视化操作）
    - 任务（具体某个场景下的操作步骤、北极星的运维手册、故障处理）
  - 参考文档（插件开发、北极星扩展）
  - 开源社区
    - 开发者社区
      - 贡献指南
      - 路线规划 

### sdk中「推空保护」的问题

- 推空保护，核心在于最终的结果是幂等的
- 推空保护的开关将挪至 provider 侧进行控制，consumer 侧默认全部开启推空保护措施


### 测试

- 部分仓库需要添加PR时自动扫描单元测试代码覆盖率的变化情况
- 优先完成单元测试
